app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +70,61 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
+            val current: Calendar = Calendar.getInstance()
+            val currentHour = current.get(Calendar.HOUR_OF_DAY)
+            val currentMinute = current.get(Calendar.MINUTE)
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            if (item.ehr == currentHour && item.emin == currentMinute
+            ) {
+                StoreSession.writeInt(AppConstants.BEGIN_STATUS, 0)
+                StoreSession.writeLong(AppConstants.ACTIVE_PROFILE_ID, 0)
+            }
+            if (item.shr == currentHour && item.smin == currentMinute
+            ) {
+                StoreSession.writeLong(AppConstants.ACTIVE_PROFILE_ID, item.profileId)
+            }
+            binding.pauseSwitch.setOnClickListener {
+                item.pauseSwitch = false
+                WorkManager.getInstance(profileViewModel.getApplication())
```

Comment:
Not as expected. Defeats the purpose of making the WorkManagerHelper class and initializing the work manager instance. Change the code so that context has not to be passed from ProfileListAdapter to make the WorkManager instance.

Reply:
I have made the correction in the code

Reply:
@rob729 One more thing, the following function also Defeats the purpose of making the WorkManagerHelper class and initializing the work manager instance. 

 fun removeWork(tag: String) {
        WorkManager.getInstance(profileViewModel.getApplication()).cancelAllWorkByTag(tag)
    }

Since I had not made any changes in these lines for this PR that's why I did not change it. Shall I modify it also?

Reply:
Yes @coder2699

Reply:
@rob729 I have made the changes, but not pushed the code yet. Please let me know if there are any other changes, like in variable names or something.

Reply:
@rob729 I have pushed the modified code